packages/apple-pay-integration/e2e/applepay.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { test, CustomerStepPreset, PaymentStepAsGuestPreset } from '@bigcommerce/checkout/payment-integration-test-framework';
+import { applePayCart, internalOrder, order, orderPayment, validateMerchantResponse, consignmentsAndBilling } from './ApplePayTestMockResponse';
+import addApplePaySessionToChromePaymentStep from './ApplePaySessionPaymentStepMockObject'
+import addApplePaySessionToChromeCustomerStep from './ApplePaySessionCustomerStepMockObject'
+
+test.describe('ApplePay', () => {
+    test('Customer should be able to pay using ApplePay through the payment step in checkout', async ({ assertions, checkout, page }) => {
+        // Testing environment setup
+        await page.addInitScript(addApplePaySessionToChromePaymentStep);
+
+        const responseProps = { status: 200, contentType: 'application/json' };
+        await checkout.use(new PaymentStepAsGuestPreset());
+        await checkout.start('ApplePay in Payment Step')
+        await checkout.route(/order-confirmation.*/, './packages/payment-integration-test-framework/src/support/orderConfirmation.ejs', { orderId: '124' });
+        await page.route('**/api/storefront/payments/applepay?cartId=124', route => route.fulfill({ ...responseProps, body: applePayCart }));
+        await page.route('**/api/public/v1/payments/applepay/validate_merchant', route => route.fulfill({ ...responseProps, body: validateMerchantResponse }));
+        await page.route(/.*\/api\/storefront\/orders\/124.*/, route => route.fulfill({ ...responseProps, body: order }));
+        await page.route('**/api/public/v1/orders/payments', route => route.fulfill({ ...responseProps, body: orderPayment }));
+        await page.route('**/internalapi/v1/checkout/order', route => route.fulfill({
+            ...responseProps,
+            headers: { Token: 'White shirt now red, my bloody nose' },
+            body: internalOrder,
+        }));
+
+        // Playwright actions
+        await checkout.goto();
+        await page.locator('label div').first().click();
```

Comment:
Prefer to have a non-ambiguous selector, like `id` or `data-test`. `label div` is too ambiguous because it doesn't refer to a specific element and any changes to the HTML hierarchy could break the test.

Reply:
Currently I have used alt text to select Apple Pay, but after discussion https://bigcommerce.slack.com/archives/C02PJTFL5P0/p1661385791604789, we would also be adding `data-testid` instead to have an opinionated way of doing payment step tests